opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -86,13 +85,39 @@ public synchronized void reset() {
     }
 
     /**
-     * @return a copy of all finish()ed MockSpans started by this MockTracer (since construction or the last call to
+     * @return all finish()ed MockSpans started by this MockTracer (since construction or the last call to
      * MockTracer.reset()).
      *
      * @see MockTracer#reset()
      */
     public synchronized List<MockSpan> finishedSpans() {
-        return new ArrayList<>(this.finishedSpans);
+        return Collections.unmodifiableList(new ArrayList<>(this.finishedSpans));
```

Comment:
That's a breaking change. Client code may rely on the fact that it gets a copy right now which it can safely mutate.

Reply:
Yes, It is a breaking change.

I think it is not right to make `finishedTraces` unmodifiable while `finishedSpans` modifiable. @pavolloffay Shall we make them all modifiable?

Reply:
This should be reverted. In the other method we can return normal map for the consistency.

Reply:
> This should be reverted. In the other method we can return normal map for the consistency.

Agreed as well.

Reply:
Done.